src/Features/LanguageServer/CommonLanguageServerProtocol.Framework/ILspLogger.cs

```diff
@@ -0,0 +1,18 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+
+namespace CommonLanguageServerProtocol.Framework;
+
+// Open to suggestions about changes here, gut feeling is that these are the wrong abstractions.
```

Comment:
Ya I'm also torn. Looking at [ILogger](https://github.com/dotnet/runtime/blob/06e4c45624eebbc5ecb1e10cccca52c5838ea072/src/libraries/Microsoft.Extensions.Logging.Abstractions/src/ILogger.cs) it feels like we may be missing a good amount here but I'm not quite sure. It's definitely interesting how we're baking "trace level" concepts into the method names and that parameters aren't quite here though